src/heap.rs

```diff
@@ -433,6 +472,24 @@ impl HeapGuard {
             len: 0,
         })
     }
+
+    pub fn alloc_raw(&mut self, layout: Layout) -> Result<NonNull<()>, ()> {
+        // Clean up any pending allocs
+        self.clean_allocs();
```

Comment:
I know this is probably consistent with the other allocations, but I forget mnemos-alloc did this, and it's probably better to just handle this in a batch, like we do in mnemos kernel's `Kernel::tick`. Leave it as-is, I should make an issue to track this elsewhere.